modules/id5IdSystem.js

```diff
@@ -177,83 +242,101 @@ class IdFetchFlow {
     this.usPrivacyData = usPrivacyData
   }
 
-  execute() {
-    return this.#callForConfig(this.submoduleConfig)
-      .then(fetchFlowConfig => {
-        return this.#callForExtensions(fetchFlowConfig.extensionsCall)
-          .then(extensionsData => {
-            return this.#callId5Fetch(fetchFlowConfig.fetchCall, extensionsData)
-          })
-      })
-      .then(fetchCallResponse => {
-        try {
-          resetNb(this.submoduleConfig.params.partner);
-          if (fetchCallResponse.privacy) {
-            storeInLocalStorage(ID5_PRIVACY_STORAGE_NAME, JSON.stringify(fetchCallResponse.privacy), NB_EXP_DAYS);
-          }
-        } catch (error) {
-          logError(LOG_PREFIX + error);
-        }
-        return fetchCallResponse;
-      })
+  /**
+   * Calls the ID5 Servers to fetch an ID5 ID
+   * @returns {Promise<IdResponse>} The result of calling the server side
+   */
+  async execute() {
+    if (this.#isExternalModule()) {
+      try {
+        return this.#externalModuleFlow();
+      } catch (error) {
+        logError('Error while performing ID5 external module flow. Continuing with regular flow.', error);
+        return this.#regularFlow();
+      }
+    } else {
+      return this.#regularFlow();
+    }
   }
 
-  #ajaxPromise(url, data, options) {
-    return new Promise((resolve, reject) => {
-      ajax(url,
-        {
-          success: function (res) {
-            resolve(res)
-          },
-          error: function (err) {
-            reject(err)
-          }
-        }, data, options)
-    })
+  #isExternalModule() {
+    return typeof this.submoduleConfig.params.externalModuleUrl === 'string';
   }
 
   // eslint-disable-next-line no-dupe-class-members
-  #callForConfig(submoduleConfig) {
+  async #externalModuleFlow() {
+    const results = await GreedyPromise.allSettled([
+      this.#loadExternalModule(this.submoduleConfig.params.externalModuleUrl),
+      this.#callForConfig(this.submoduleConfig)
+    ]);
+
+    if (isRejected(results[0])) {
+      throw new Error('Could not load external module:', results[0].reason);
+    }
+    if (isRejected(results[1])) {
+      throw new Error('Could not load configuration:', results[0].reason);
```

Comment:
this probably wants to use `results[1]`.

Reply:
Thanks, good catch